### PR TITLE
busybox: add IPv6 support

### PR DESCRIPTION
**Pull request by mateuszkobak**
## Description

* Added necessary nonblocking flag for socket used in ping6, following the existing solution for ping4
* Provided IPv6 addresses info for ifconfig
* Made `route.c` compile by including `net/ip6_route.h`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic-qemu (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
https://github.com/phoenix-rtos/phoenix-rtos-lwip/pull/93
https://github.com/phoenix-rtos/libphoenix/pull/355
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: busybox/22-ifconfig-ipv6.patch [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
++						n = n/2;
++					}
++					i++;
++				}
++
++				printf("          inet6 addr: %s/%d", addr6, j);
++				printf(" Scope:");
++
++				if (IN6_IS_ADDR_LINKLOCAL(&addr->sin6_addr)) {
++					puts ("link-local");
```
Comment (nalajcie) [inline busybox/22-ifconfig-ipv6.patch:128]: space after `puts` is non-standard even for Busybox coding style

File: busybox/22-ifconfig-ipv6.patch [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ #define _PATH_PROCNET_DEV               "/proc/net/dev"
+-#define _PATH_PROCNET_IFINET6           "/proc/net/if_inet6"
+ 
+ #ifdef HAVE_AFINET6
+ # ifndef _LINUX_IN6_H
+@@ -841,6 +841,7 @@ static void print_bytes_scaled(unsigned long long ull, const char *end)
+ 
+ 
+ #ifdef HAVE_AFINET6
++#if 0
```
Comment (nalajcie) [inline busybox/22-ifconfig-ipv6.patch:88]: maaybe we could use eg. `!defined(__phoenix__)` to highlight that this is our custom implementation (while viewing code after patching)?

**Review by nalajcie — COMMENTED** [review]
small nitpicks, otherwise LGTM.

**Review by nalajcie — APPROVED** [review]

**Comment by anglov** [comment]
Please rebase

**Comment by mateuszkobak** [comment]
Rebased

File: busybox/22-ifconfig-ipv6.patch [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
++	while (ifap != NULL) {
++		if (strcmp(ifap->ifa_name, ptr->name) == 0) {
++			struct sockaddr_in6 *addr;
++			addr = ifap->ifa_addr;
++
++			if (addr->sin6_family == AF_INET6) {
++				inet_ntop(AF_INET6, &addr->sin6_addr, addr6, 50);
++
++				/* Get the prefix length */
++				unsigned char *c = ((struct sockaddr_in6 *)ifap->ifa_netmask)->sin6_addr.s6_addr;
```
Comment (badochov) [inline busybox/22-ifconfig-ipv6.patch:112]: This breaks: `-Wdeclaration-after-statement` declared in `Makefile.flags` of busybox. For some reason on GCC 9.5 warn is not issued but on gcc 14.2 it is.

File: busybox/22-ifconfig-ipv6.patch [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+ 							}
+ 							if (lsa->u.sa.sa_family == AF_INET6) {
+ 								int sockfd6;
+-								struct in6_ifreq ifr6;
++								struct in6_aliasreq ifr6;
+ 
+ 								sockfd6 = xsocket(AF_INET6, SOCK_DGRAM, 0);
+-								xioctl(sockfd6, SIOCGIFINDEX, &ifr);
+-								ifr6.ifr6_ifindex = ifr.ifr_ifindex;
+-								ifr6.ifr6_prefixlen = prefix_len;
```
Comment (badochov) [inline busybox/22-ifconfig-ipv6.patch:53]: This break `-Wunused-but-set-variable` on prefix_len which is part of `-Wall`

File: busybox/22-ifconfig-ipv6.patch (hunk 3 above)
Comment (badochov) [reply busybox/22-ifconfig-ipv6.patch:112]: Update its also triggered on 9.50 my bad

**Pull request merged**